src/mixins/createFormLabelMixin.ts

```diff
@@ -0,0 +1,164 @@
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import compose, { ComposeFactory } from '@dojo/compose/compose';
+import { assign } from '@dojo/core/lang';
+import { NodeAttributeFunction, DNode, Widget, WidgetOptions, WidgetState, WidgetProperties, FormLabelProperties } from './../interfaces';
+import { v } from '../d';
+
+export interface FormLabelMixinProperties extends WidgetProperties {
+	/**
+	 * The form widget's name
+	 */
+	name?: string;
+
+	/**
+	 * The current value
+	 */
+	value?: string;
+
+	/**
+	 * The type of the form field (equates to the `type` attribute in the DOM)
+	 */
+	type?: string;
+
+	/**
+	 * Prevents the user from interacting with the form field
+	 */
+	disabled?: boolean;
+
+	/**
+	 * Label settings for form label text, position, and visibility
+	 */
+	label?: string | FormLabelProperties;
+
+	/**
+	 * Accessibility attributes
+	 */
+	checked?: boolean;
+	descriptionID?: string;
+	inputmode?: string;
+	invalid?: boolean;
+	maxlength?: number | string;
+	minlength?: number | string;
+	multiple?: boolean;
+	placeholder?: string;
+	readonly?: boolean;
+	required?: boolean;
+}
+
+export interface FormLabelMixin {
+	/**
+	 * A function that generates node attribtues for the child form field
+	 */
+	getFormFieldNodeAttributes: NodeAttributeFunction<this>;
+
+	/**
+	 * Override the default getNode function on createWidgetBase to return child input
+	 */
+	getNode(): DNode;
+}
+
+export type FormLabel = Widget<FormLabelMixinProperties> & FormLabelMixin;
```

Comment:
I think we should provide doc on all exported interfaces, types, functions, etc.